app/api/cloud_account/cloud_accounts_manager.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import logging
+
+from fastapi import status as http_status
+
+from app.api.cloud_account.cloud_accounts_conf.aws import AWSConfigStrategy
+from app.api.cloud_account.cloud_accounts_conf.azure import (
+    AzureCNRConfigStrategy,
+    AzureTenantConfigStrategy,
+)
+from app.api.cloud_account.cloud_accounts_conf.cloud_config_strategy import (
+    CloudConfigStrategy,
+)
+from app.api.cloud_account.cloud_accounts_conf.gcp import GCPCNRConfigStrategy
+from app.core.exceptions import (
+    CloudAccountConfigError,
+    OptScaleAPIResponseError,
+)
+from app.optscale_api.cloud_accounts import OptScaleCloudAccountAPI
+
+logger = logging.getLogger(__name__)
+
+
+class CloudStrategyConfiguration:
+    ALLOWED_PROVIDERS = {
+        "aws_cnr": AWSConfigStrategy,
+        "gcp_cnr": GCPCNRConfigStrategy,
+        "azure_cnr": AzureCNRConfigStrategy,
+        "azure_tenant": AzureTenantConfigStrategy,
+    }
+
+    def __init__(
+        self,
+        name: str,
+        provider_type: str,
+        auto_import: bool = True,
+        process_recommendations: bool = True,  # noqa: E501
+        config: dict = None,
+    ):
+        self.name = name
+        self.type = provider_type
+        self.config = config if config else {}
+        self.auto_import = auto_import
+        self.process_recommendations = process_recommendations
+
+    def select_strategy(self):
+        """
+        This method checks if the Cloud Account type is allowed.
+        If it's valid, an instance of the Cloud Account Class's Strategy
+        will be created and validated.
+        :return:
+        :rtype:
+        """
+        if self.type not in self.ALLOWED_PROVIDERS:
+            raise OptScaleAPIResponseError(
+                title="Wrong Cloud Account",
+                error_code="OE0436",
```

Comment:
from were this code is coming from ? Same they raise if you put an invalid value ?